File: win-prog/ProgWinCS/Brushes and Pens/TwoPointLinearGradientBrush/TwoPointLinearGradientBrush.cs
//----------------------------------------------------------
// TwoPointLinearGradientBrush.cs © 2001 by Charles Petzold
//----------------------------------------------------------
using System;
using System.Drawing;
using System.Drawing.Drawing2D;
using System.Windows.Forms;

class TwoPointLinearGradientBrush: PrintableForm
{
     public new static void Main()
     {
          Application.Run(new TwoPointLinearGradientBrush());
     }
     TwoPointLinearGradientBrush()
     {
          Text = "Two-Point Linear Gradient Brush";
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          LinearGradientBrush lgbrush = 
               new LinearGradientBrush(
                         new Point(cx / 4, cy / 4), 
                         new Point(3 * cx / 4, 3 * cy / 4),
                         Color.White, Color.Black);

          grfx.FillRectangle(lgbrush, 0, 0, cx, cy);
     }
}

File: win-prog/ProgWinVB/Brushes and Pens/TwoPointLinearGradientBrush/TwoPointLinearGradientBrush.vb
'------------------------------------------------------------
' TwoPointLinearGradientBrush.vb (c) 2002 by Charles Petzold
'------------------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Drawing.Drawing2D
Imports System.Windows.Forms

Class TwoPointLinearGradientBrush
    Inherits PrintableForm

    Shared Shadows Sub Main()
        Application.Run(New TwoPointLinearGradientBrush())
    End Sub

    Sub New()
        Text = "Two-Point Linear Gradient Brush"
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        Dim lgbr As New LinearGradientBrush( _
                                New Point(cx \ 4, cy \ 4), _
                                New Point(3 * cx \ 4, 3 * cy \ 4), _
                                Color.White, Color.Black)
        grfx.FillRectangle(lgbr, 0, 0, cx, cy)
    End Sub
End Class
